lib/go/test/flow_dkg_test.go

```diff
@@ -566,8 +566,76 @@ func TestDKG(t *testing.T) {
 		)
 	})
 
+	t.Run("Should not be able to set the safe threshold while the DKG is enabled", func(t *testing.T) {
+
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateSetSafeThresholdScript(env), DKGAddress)
+
+		err := tx.AddArgument(CadenceUFix64("0.90"))
+		require.NoError(t, err)
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, DKGAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), DKGSigner},
+			true,
+		)
+
+	})
+
+	t.Run("Admin Should be able to force stop the DKG", func(t *testing.T) {
+
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateForceStopDKGScript(env), DKGAddress)
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, DKGAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), DKGSigner},
+			false,
+		)
+	})
+
+	t.Run("Should be able to set the safe threshold while the DKG is disabled", func(t *testing.T) {
```

Comment:
Could we also have a test for:
* setting `safeThreshold > 1` should revert
* setting `safeThreshold < 0.5` should result in the "native threshold" being used

Reply:
I just included in the pre-condition for the setSafeThreshold function that it has to be > 0.5